core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/filter/SpecializedFilterOpExec.java

```diff
@@ -16,12 +18,20 @@ public SpecializedFilterOpExec(SpecializedFilterOpDesc opDesc) {
                 (Function1<Tuple, Boolean> & Serializable) this::filterFunc);
     }
 
+    // MARK: Debug/unittests only
+    public SpecializedFilterOpExec(SpecializedFilterOpDesc opDesc, List<FilterPredicate> predicates) {
+        this(opDesc);
+        opDesc.predicates = predicates;
+    }
+
     public Boolean filterFunc(Tuple tuple) {
         boolean satisfy = false;
         for (FilterPredicate predicate : opDesc.predicates) {
-            satisfy = satisfy || predicate.evaluate(tuple, opDesc.context());
+            if (predicate.evaluate(tuple, opDesc.context())) {
+                return true;
+            }
```

Comment:
what is this change for?

Reply:
If any of the predicates are ever true, we never need to do any more iterations, since a tuple matches. It's equivalent to the previous code since satisfy || and return satisfy at the end is only going to be true if any of the predicates are true.

Reply:
Changed it to stream anyMatch because that's exactly what we're doing here.